### PR TITLE
Super header

### PR DESCRIPTION
**Pull request by rossbowen**
image

For #293: bases the climate change header on the GOVUK header. Have reworked code from the `govuk_publishing_components` repo:
- [scss styling](https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/stylesheets/govuk_publishing_components/components/_layout-super-navigation-header.scss)
- [html (written in ruby)](https://github.com/alphagov/govuk_publishing_components/blob/main/app/views/govuk_publishing_components/components/_layout_super_navigation_header.html.erb)
- [javascript](https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/javascripts/govuk_publishing_components/components/layout-super-navigation-header.js)

We produce a [`SuperNavigationHeader`](https://github.com/GSS-Cogs/dd-cms/blob/super-header/volto/src/addons/volto-climatechange-elements/src/components/CcSuperNavigationHeader/CcSuperNavigationHeader.jsx) component which takes a [JSON input](https://github.com/GSS-Cogs/dd-cms/blob/super-header/volto/src/addons/volto-climatechange-elements/src/customizations/volto/components/theme/Header/Header.jsx#L11-L84) which structures the behaviour of the nav.

The JSON input is hardcoded. I'd like to use the [state](https://github.com/GSS-Cogs/dd-cms/blob/super-header/volto/src/addons/volto-climatechange-elements/src/customizations/volto/components/theme/Header/Header.jsx#L130) and wrangle this into the input, but it doesn't currently give me all the information I'd need. Can create a new issue. ATM it returns:

```js
[
  { title: 'Home', url: '', items: [] },
  { title: 'Dashboards', url: '/dashboards', items: [] },
  { title: 'Articles', url: '/articles', items: [] }
]
```

### CODE REVIEW

File: volto/src/addons/volto-climatechange-elements/src/components/CcSuperNavigationHeader/components/SuperNavigationSearchMenu.jsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+                                <div
+                                    className="gem-c-search govuk-!-display-none-print gem-c-search--large gem-c-search--on-white gem-c-search--separate-label"
+                                    data-module="gem-toggle-input-class-on-focus"
+                                    data-gem-toggle-input-class-on-focus-module-started="true"
+                                >
+                                    <label htmlFor="search-main-1d4ac258" className="govuk-label govuk-label--m">{props.search_text}</label>
+                                    <div className="gem-c-search__item-wrapper">
+                                        <input
+                                            enterkeyhint="search"
```
Comment (anojperera) [inline volto/src/addons/volto-climatechange-elements/src/components/CcSuperNavigationHeader/components/SuperNavigationSearchMenu.jsx:108]: can we change this to `enterKeyHint`

File: volto/src/addons/volto-climatechange-elements/src/components/CcSuperNavigationHeader/components/SuperNavigationSearchMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                                            id="search-main-1d4ac258"
+                                            name="q"
+                                            title="Search"
+                                            type="search"
+                                        />
+                                        <div className="gem-c-search__item gem-c-search__submit-wrapper">
+                                            <button
+                                                className="gem-c-search__submit"
+                                                type="submit"
+                                                enterkeyhint="search"
```
Comment (anojperera) [inline volto/src/addons/volto-climatechange-elements/src/components/CcSuperNavigationHeader/components/SuperNavigationSearchMenu.jsx:119]: Same as above `enterKeyHint`
There's an annoying warning...

**Review by anojperera — APPROVED** [review]
If you incorporate the couple of changes it will get rid of the warnings

**Review by ajtucker — APPROVED** [review]
Looks good to go, though would be good to fix the warnings @anojperera noted and the `[object.object]` href.

**Pull request merged**